crates/mun_language_server/src/main_loop.rs

```diff
@@ -37,14 +231,337 @@ enum LoopState {
     Shutdown,
 }
 
+/// Handles a received request
+async fn handle_request(request: Request, connection: &mut ConnectionState) -> Result<LoopState> {
+    if connection.connection.handle_shutdown(&request).await? {
+        return Ok(LoopState::Shutdown);
+    };
+    Ok(LoopState::Continue)
+}
+
+/// Handles a received notification
+async fn on_notification(
+    notification: Notification,
+    connection: &mut ConnectionState,
+    state: &LanguageServerState,
+) -> Result<LoopState> {
+    let notification =
+        // When a a text document is opened
+        match cast_notification::<lsp_types::notification::DidOpenTextDocument>(notification) {
+            Ok(params) => {
+                // Get the uri
+                let uri = params.text_document.uri;
+                // And convert into a file path
+                let path = uri
+                    .to_file_path()
+                    .map_err(|()| anyhow!("invalid uri: {}", uri))?;
+                if state
+                    .vfs
+                    .write()
+                    .await
+                    .add_file_overlay(&path, params.text_document.text).is_some()
+                {
+                    //loop_state.subscriptions.add_sub(FileId(file_id.0));
```

Comment:
What's the reason this was commented? If it is not necessary, can it be removed altogether?

Reply:
@tdejager ?

Reply:
Replaced with
```
// TODO: Keep track of opened files
```